src/test/java/edu/ucsb/courses/controllers/ScheduleItemControllerTests.java

```diff
@@ -0,0 +1,487 @@
+
+import javax.swing.text.html.Option;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestPostProcessors.csrf;
+
+
+// @Import(SecurityConfig.class) applies the security rules 
+// so that /api/public/** endpoints don't require authentication.
+// Otherwise you may get authorization errors when running the test
+
+@WebMvcTest(value = ScheduleItemController.class)
+@WithMockUser
+//@Import(SecurityConfig.class)
+public class ScheduleItemControllerTests {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @MockBean
+    private ScheduleItemRepository scheduleItemRepository;
+
+    @MockBean
+    private ScheduleRepository scheduleRepository;
+
+    @MockBean
+    private ArchivedCourseRepository courseRepo;
+
+    private String userToken = "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTYiLCJuYW1lIjoiSm9obiBEb2UiLCJpYXQiOjE1MTYyMzkwMjJ9.MkiS50WhvOFwrwxQzd5Kp3VzkQUZhvex3kQv-CLeS3M";
+
+    private ObjectMapper objectMapper = new ObjectMapper();
+
+    @Test
+    public void test_getScheduleItemSuccess() throws Exception {
+
+        String expectedResult = "{id=1 courseId='CS 156', enrollCode='Adv App Programming', scheduleId='1}";
+        String urlTemplate = "/api/public/getScheduleItemById?id=%s";
+        String url = String.format(urlTemplate, "1");
+
+        DecodedJWT jwt = JWT.decode(userToken.substring(7));
+
+        ScheduleItem scheduleItem = new ScheduleItem(1L,"CS 156", "Adv App Programming", jwt.getSubject(), 1L);
+
+        Optional<ScheduleItem> opt = Optional.of(scheduleItem);
+
+        when(scheduleItemRepository.findById(any(Long.class)))
+                .thenReturn(opt);
+
+        MvcResult response = mockMvc.perform(get(url).contentType("application/json").header(HttpHeaders.AUTHORIZATION,userToken)).andExpect(status().isOk())
+                .andReturn();
+
+        String responseString = response.getResponse().getContentAsString();
+        ScheduleItem returnVal = objectMapper.readValue(responseString,ScheduleItem.class);
+
+        assertEquals(scheduleItem, returnVal);
+    }
+
+
+    @Test
+    public void test_getScheduleFailureNoScheduleItem() throws Exception {
+        String expectedResult = "";
+        String urlTemplate = "/api/public/getScheduleItemById?id=%s";
+        String url = String.format(urlTemplate, "1");
+
+        //Schedule schedule = new Schedule(1L,"CS 156", "Adv App Programming", "Fall 2020", "s");
```

Comment:
We can remove unused code instead of commenting it out.

Reply:
Yup